containerd/main.go

```diff
@@ -66,6 +71,7 @@ func main() {
 	setAppBefore(app)
 
 	app.Action = func(context *cli.Context) {
+		os.Setenv("CONTAINERD_SHIM", context.String("shim"))
```

Comment:
Could you instead add it to the supervisor and container type (the same way it was done for the runtime)?